### ISSUE TITLE
Persistent journald rotation prevents rsyslog from reading some messages via imjournal

### ISSUE THREAD

**Issue by onyxmaster** [rsyslog/rsyslog#3732]
### Expected behavior
The journald file rotation does not affect log receiving.

### Actual behavior
Occasionaly when journald rotates its persistent storage (for example due to max size being reached), the imjournal module stops receiving data from per-user unit journal (`journalctl --user`). There are no specific errors logged apart of the usual (normal) line:
```
Jul  1 06:45:07 fornax2 rsyslogd[69136]: imjournal: journal reloaded... [v8.1905.0 try https://www.rsyslog.com/e/0 ]
```

### Steps to reproduce the behavior
I undestand that this issue description is very vague and I can't reliably reproduce it, but it happened at least 3 times in the last week. The "system" journald appear to pass through nicely no matter the reloads, but the "user" ones do not process reliably. Restarting rsyslog or waiting until it reloads the imjournal again helps, but the events are quite often not processed until this happens. As far as I know, journald stores user and system events separately, maybe the open cursor does not track properly all these streams due to a bug in journald?

### Environment
- rsyslog version: 8.1905.0
- platform: Ubuntu 16.04
- systemd: 229

```
module(load="imjournal"
    StateFile="imjournal.state"
    IgnorePreviousMessages="on"
    PersistStateInterval="100"
    WorkAroundJournalBug="on"
    RateLimit.Interval="3"
    RateLimit.Burst="3333")
```

**Comment by rgerhards** [comment]
I would assume this is a bug in the systemd-journal library. @jvymazal  what do you think?

**Comment by jvymazal** [comment]
the sytemd/journald internals are opaque to outside world so if some logs are getting trough and others not I am afraid there is nothing I can do about it.

**Comment by rgerhards** [comment]
> the sytemd/journald internals are opaque to outside world so if some logs are getting trough and others not I am afraid there is nothing I can do about it.

I interpret this as "it's a systemd issue"?  just to not mistakenly think so...

**Comment by jvymazal** [comment]
And the reloaded line should not happen after #3566 which is strange. @onyxmaster would you have debug logs from one of the occasions this happened?

**Comment by jvymazal** [comment]
@rgerhards most probably yes

**Comment by rgerhards** [comment]
@onyxmaster can you run `journalctl` in follow mode? Often it experiences exactly the same issue, and that is easier to report to the journal folks.

**Comment by onyxmaster** [comment]
@rgerhards excellent suggestion, I'll leave it running on the suspect servers for some time and will report my findings later.

**Comment by rgerhards** [comment]
@onyxmaster excellent!

**Comment by rgerhards** [comment]
Apologies for the very late reply. This is probably no longer relevant to the original reporter, but I am adding the conclusion for people who arrive here via search while we work through the backlog. We do not close rsyslog issues lightly, and we are not closing this merely because of age.

Triage result: not reproducible / needs reporter info.

Reasoning: The issue appeared likely to be journald-side rotation/follow behavior. The maintainer suggested comparing with `journalctl -f`, the reporter agreed, and no rsyslog-side evidence followed.

If this still reproduces on a current rsyslog release and maintained platform, please open a new issue with the current version, complete configuration, logs, and a minimal reproducer so we can investigate it as a fresh current problem.
